src/tests/multihost/ipa/test_adtrust.py

```diff
@@ -528,3 +529,85 @@ def test_idview_override_group_fails(multihost, create_aduser_group):
             "Group 2 id was not overridden."
         assert f"domain users@{domain}" in cmd.stdout_text, \
             "Group domain users is missing."
+
+    @staticmethod
+    @pytest.mark.stv
+    def test_ad_user_ssh_ipa_client(multihost):
+        """
+        :title: Cannot SSH with AD user to ipa-client
+        :description: 'krb5_validate' and 'pac_check' settings conflicted
+          before the fix. By default 'krb5_validate = true' with id_providers
+          ipa and ad_provider. Setting it to false will enable login even if
+          the ticket validation does not work
+        :id: d4af4084-0ee2-4339-be66-ef117439f32d
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2128544
+                   https://bugzilla.redhat.com/show_bug.cgi?id=2128902
+        :steps:
+          1. Add an invalid entry in keytab after taking a backup
+          2. ssh as AD user on the ipa client machine
+          3. Set 'krb5_validate = false', debug_level 9 in sssd.conf,
+             and restart sssd service
+          4. ssh as AD user on the ipa client machine
+          5. Check for log message on PAC check being skipped with
+             krb_validate set to false
+        :expectedresults:
+          1. Invalid entry is added at the end of the keytab
+          2. AD user ssh login fails
+          3. sssd restart successfully with 'krb5_validate = false'
+          4. AD user ssh login is successful
+          5. Log message about skipping pac_check is found in domain log and
+             krb5_child.log
+        """
+        ad_domain_name = multihost.ad[0].domainname
+        client_hostip = multihost.client[0].ip
+        multihost.client[0].run_command('cp -f /etc/krb5.keytab ' \
+                                        '/etc/krb5.keytab.bak', raiseonerr=False)
+        ktutil_cmd = '(echo "rkt /etc/krb5.keytab"; sleep 1; echo "addent ' \
+                     '-password -p invalid@invaliddom -k 3 -e ' \
+                     'aes128-cts-hmac-sha1-96"; sleep 1; echo "pass00189";' \
+                     'sleep 1; echo "list"; sleep 1; echo ' \
+                     '"wkt /tmp/invalid.keytab"; sleep 1; echo "quit";) | ktutil'
+
+        multihost.client[0].run_command(ktutil_cmd, raiseonerr=False)
+        multihost.client[0].run_command('mv /tmp/invalid.keytab ' \
```

Comment:
Unnecessary backslash in expression. These can be also seen in flack8 errors: ./src/tests/multihost/ipa/test_adtrust.py:563:67: E502 the backslash is redundant between brackets